ras_backstage/resources/reporting_units/get_reporting_unit.py

```diff
@@ -71,6 +72,15 @@ def add_collection_exercise_details(collection_exercises, reporting_unit, cases)
 def link_respondents_to_survey(respondents, survey):
     survey['respondents'] = []
     for respondent in respondents:
+
+        # Collect the collection case.
+        json = get_cases_by_business_party_id(respondent.get('id'))
+        try:
+            respondent['case_id'] = json[0].get('id')
```

Comment:
This can also throw a KeyError if cases[0] is empty for whatever reason.  We need to be handling that exception too